demos/face_recognition_demo/python/face_detector.py

```diff
@@ -0,0 +1,99 @@
+"""
+ Copyright (c) 2018 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+from utils import resize_input
+from ie_module import Module
+
+
+class FaceDetector(Module):
+    class Result:
+        OUTPUT_SIZE = 7
+
+        def __init__(self, output):
+            self.image_id = output[0]
+            self.label = int(output[1])
+            self.confidence = output[2]
+            self.position = np.array((output[3], output[4])) # (x, y)
+            self.size = np.array((output[5], output[6])) # (w, h)
+
+        def rescale_roi(self, roi_scale_factor=1.0):
+            self.position -= self.size * 0.5 * (roi_scale_factor - 1.0)
+            self.size *= roi_scale_factor
+
+        def resize_roi(self, frame_width, frame_height):
+            self.position[0] *= frame_width
+            self.position[1] *= frame_height
+            self.size[0] = self.size[0] * frame_width - self.position[0]
+            self.size[1] = self.size[1] * frame_height - self.position[1]
+
+        def clip(self, width, height):
+            min = [0, 0]
+            max = [width, height]
+            self.position[:] = np.clip(self.position, min, max)
+            self.size[:] = np.clip(self.size, min, max)
+
+    def __init__(self, ie, model, input_height, input_width, confidence_threshold=0.5, roi_scale_factor=1.15):
+        super(FaceDetector, self).__init__(ie, model)
+
+        if input_height and input_width:
```

Comment:
The `input_height` and `input_width` should be tested for positivity somewhere.

Reply:
Updated.